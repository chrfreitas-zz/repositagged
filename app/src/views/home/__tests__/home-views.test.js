import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Home from '../index';

describe('Home view', () => {
  it('should match with snapshot', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Home />);
    expect(result).toMatchSnapshot();
  });

  it('should change state username', () => {
    const evt = {
      target: { value: 'tarantino' },
    };

    const wrapper = shallow(<Home />);
    wrapper.instance().onChangeInput(evt);

    expect(wrapper.state('username')).toBe(evt.target.value);
  });
});
