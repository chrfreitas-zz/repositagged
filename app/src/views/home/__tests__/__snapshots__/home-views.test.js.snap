// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view should match with snapshot 1`] = `
<div
  className="home"
>
  <div
    className="home__row"
  >
    https://github.com/
    <Input
      placeholder="username"
    />
  </div>
  <div
    className="home__row"
  >
    <Button>
      get repositories
    </Button>
  </div>
</div>
`;
